api/v1beta1/azuremachine_default.go

```diff
@@ -67,7 +68,12 @@ func (s *AzureMachineSpec) SetDataDisksDefaults() {
 			}
 		}
 		if disk.CachingType == "" {
-			s.DataDisks[i].CachingType = "ReadWrite"
+			if s.DataDisks[i].ManagedDisk != nil &&
+				s.DataDisks[i].ManagedDisk.StorageAccountType == string(compute.StorageAccountTypesUltraSSDLRS) {
+				s.DataDisks[i].CachingType = "None"
```

Comment:
nit: let's take advantage of this change to use the consts `compute.CachingTypesNone` and `compute.CachingTypesReadWrite` here

Reply:
That's a good idea. Done 👍